filters/openpolicyagent/openpolicyagent.go

```diff
@@ -292,9 +303,9 @@ func (registry *OpenPolicyAgentRegistry) newOpenPolicyAgentInstance(bundleName s
 		return nil, err
 	}
 
-	err = engine.waitPluginsReady(100*time.Millisecond, 30*time.Second)
+	err = engine.waitPluginsReady(100*time.Millisecond, config.startupTimeout)
 	if err != nil {
-		engine.Logger().WithFields(map[string]interface{}{"err": err}).Error("Failed to wait for plugins activation.")
+		engine.Close(ctx)
```

Comment:
I am wondering if we should use https://pkg.go.dev/context#WithTimeout for `engine.Start(ctx)` instead of `context.Background()`

Reply:
Note that if context has expired, `engine.Close(ctx)` would likely fail.
Do we even need to Close if it failed to start?

Reply:
Within `engine.Start`, there's a faster function that starts the engine officially. However we need some extra opa plugins to be ready in order for the started engine to be useful. That extra wait for the plugins is likely to exceed the deadline for various error scenarios. Even when the deadline is met, engine is started with non ready plugins therefore had to be closed explicitly. But let me try to make it a bit better